examples/ShowPlayer.cpp

```diff
@@ -66,90 +69,184 @@ int ShowPlayer::Init() {
   return ola::EXIT_OK;
 }
 
+
 int ShowPlayer::Playback(unsigned int iterations,
-                         unsigned int duration,
-                         unsigned int delay) {
+                         PlaybackTime duration,
+                         PlaybackTime delay,
+                         PlaybackTime start,
+                         PlaybackTime stop) {
   m_infinite_loop = iterations == 0 || duration != 0;
   m_iteration_remaining = iterations;
   m_loop_delay = delay;
-  SendNextFrame();
+  m_start = start;
+  m_stop = stop;
 
   ola::io::SelectServer *ss = m_client.GetSelectServer();
-
   if (duration != 0) {
     ss->RegisterSingleTimeout(
-        duration * 1000,
-        ola::NewSingleCallback(ss, &ola::io::SelectServer::Terminate));
+       duration * 1000,
+       ola::NewSingleCallback(ss, &ola::io::SelectServer::Terminate));
+  }
+
+  if ((SeekTo(m_start) != ShowLoader::State::OK)) {
+    return ola::EXIT_DATAERR;
   }
   ss->Run();
   return ola::EXIT_OK;
 }
 
-void ShowPlayer::SendNextFrame() {
-  DmxBuffer buffer;
-  unsigned int universe;
-  ShowLoader::State state = m_loader.NextFrame(&universe, &buffer);
-  switch (state) {
-    case ShowLoader::END_OF_FILE:
-      HandleEndOfFile();
-      return;
-    case ShowLoader::INVALID_LINE:
-      m_client.GetSelectServer()->Terminate();
-      return;
-    default:
-      {}
+
+/**
+ * Begin playback from start point
+ */
+void ShowPlayer::Loop() {
+  ShowLoader::State state = SeekTo(m_start);
+  if (state != ShowLoader::State::OK) {
+    m_client.GetSelectServer()->Terminate();
   }
+}
 
-  state = RegisterNextTimeout();
 
-  OLA_INFO << "Universe: " << universe << ": " << buffer.ToString();
-  ola::client::SendDMXArgs args;
-  m_client.GetClient()->SendDMX(universe, buffer, args);
-
-  switch (state) {
-    case ShowLoader::END_OF_FILE:
-      HandleEndOfFile();
-      return;
-    case ShowLoader::INVALID_LINE:
-      m_client.GetSelectServer()->Terminate();
-      return;
-    default:
-      {}
+/**
+ * Seek to @p seek_time in the show file
+ * @param seek_time the time (in milliseconds) to seek to
+ */
+ShowLoader::State ShowPlayer::SeekTo(PlaybackTime seek_time) {
+  // Seeking to a time before the playhead's position requires moving from the
+  // beginning of the file.  This could be optimized more if this happens
+  // frequently.
+  if (seek_time < m_playback_pos) {
+    m_loader.Reset();
+    m_playback_pos = 0;
+  }
```

Comment:
There be dragons...

This code is only hit when looping, and I can't imagine any other situation where it would be used right now, so this is really just an acknowledgement of a potential performance issue if a hero decides to add jumps to support their Christmas light show.